peep.c

```diff
@@ -3588,6 +3588,45 @@ Perl_rpeep(pTHX_ OP *o)
                 S_check_for_bool_cxt(o, 1, OPpTRUEBOOL, 0);
             /* FALLTHROUGH */
         case OP_COND_EXPR:
+            if (o->op_type == OP_COND_EXPR) {
+                /* Is there an empty "if" block or ternary true branch?
+                   If so, optimise away the OP_STUB if safe to do so. */
+                if (cLOGOP->op_other->op_type == OP_STUB &&
+                    ((cLOGOP->op_other->op_flags & OPf_WANT) != OPf_WANT_SCALAR)
+                ) {
+                    OP *stub = cLOGOP->op_other;
+                    OP *trueop = OpSIBLING( cLOGOP->op_first );
+
+                    assert((stub == trueop ) || (OP_TYPE_IS(trueop, OP_SCOPE) &&
+                     ((stub == cUNOPx(trueop)->op_first)) && !OpSIBLING(stub))
+                    );
+                    assert(!(stub->op_flags & OPf_KIDS));
+
+                    cLOGOP->op_other = (stub->op_next == trueop) ?
+                                  stub->op_next->op_next  :
+                                  stub->op_next;
+
+                    op_sibling_splice(o, cLOGOP->op_first, 1, NULL);
+
+                    if (stub != trueop) op_free(stub);
+                    op_free(trueop);
+                } else
+
+                /* Is there an empty "else" block or ternary false branch?
+                   If so, optimise away the OP_STUB if safe to do so. */
+                if (o->op_next->op_type == OP_STUB &&
+                    ((o->op_next->op_flags & OPf_WANT) != OPf_WANT_SCALAR)
+                ) {
+                    OP *stub = o->op_next;
```

Comment:
move this line that does the assignment/caching to auto var upwards by 3 lines, see previous comment